### PR TITLE
Fix float parsing

### LINKED ISSUE #186: [bug] Wrong float constants parsing
`float` constant parsing is performed wrong.  
```
+package sandbox
+alias sprintf org.eolang.txt.sprintf
+alias stdout org.eolang.io.stdout

[args...] > app
  stdout > @
    sprintf
      "%%.9f\n"
      9999999999.0000012

```
Here, the `9999999999.0000012` is parsed as `1.0E10`:
**target/parse/app.eo.xml**
```xml

<objects>
    <o line="5" name="app">
       <o line="5" name="args" vararg=""/>
       <o base="stdout" line="6" name="@">
          <o base="sprintf" line="7">
             <o base="string" data="string" line="8">%%.9f\n</o>
             <o base="float" data="float" line="9">1.0E10</o>
          </o>
       </o>
    </o>
</objects>
```

### PR DESCRIPTION
**Pull request by nlchar**
Fixes #186

### CODE REVIEW

**Comment by nlchar** [comment]
@yegor256 could you help me, please? 
For some reasons, the tests fail with very ambiguous messages:
```
[ERROR] Failures: 
[ERROR]   PacksTest.testPacks:45 
Expected: an empty collection
     but: <[//o[@base='float' and @name='largestNegativeFloat' and @data and @text()='-1.7976931348623157E+308'], //o[@base='float' and @name='largestPositiveFloat' and @data and @text()='1.7976931348623157E+308'], //o[@base='float' and @name='smallestNegativeNormalizedFloat' and @data and @text()='-2.2250738585072014E-308'], //o[@base='float' and @name='smallestPositiveNormalizedFloat' and @data and @text()='2.2250738585072014E-308'], //o[@base='float' and @name='smallestNegativeDenormalizedFloat' and @data and @text()='-4.9E-324'], //o[@base='float' and @name='smallestPositiveDenormalizedFloat' and @data and @text()='4.9E-324']]>
[INFO] 
[ERROR] Tests run: 24, Failures: 1, Errors: 0, Skipped: 0
```
What am I doing wrong?

**Comment by yegor256** [comment]
@nlchar you should fix this line, to make it understand floats/doubles: https://github.com/cqfn/eo/blob/master/eo-parser/src/main/antlr4/org/eolang/parser/Program.g4#L240

**Comment by nlchar** [comment]
> @nlchar you should fix this line, to make it understand floats/doubles: https://github.com/cqfn/eo/blob/master/eo-parser/src/main/antlr4/org/eolang/parser/Program.g4#L240

The actual question was about how to write the XPath test properly. I investigated deeper and learned that  `text()="something"` should be used instead of `@text()="something"`.

Now tests run properly after the bug fix and fail if I undo the bug fix back to float parsing.

**Comment by nlchar** [comment]
This bug fix covers the issue of parsing EO `float` as Java `float`. Actually, EO `float` should be interpreted as Java `double`.  
If we want to add the scientific `float` notation like `1.234453E100`, just let me know and we'll try to cover it in a new issue.

**Comment by nlchar** [comment]
@yegor256 ready for review

File: sandbox/eo/app.eo [hunk 1]
```diff
@@ -6,6 +6,5 @@
 [args...] > app
   stdout > @
     sprintf
-      "%dth Fibonacci number is %d\n"
-      (args.get 0).toInt > n
-      fibonacci n
+      "%f\n"
```
Comment (yegor256) [inline sandbox/eo/app.eo:9]: @nlchar this change here is by mistake, right?

**Comment by yegor256** [comment]
@nlchar I think you forgot to revert back your changes to `app.eo`?

File: sandbox/eo/app.eo (hunk 1 above)
Comment (nlchar) [reply sandbox/eo/app.eo]: @yegor256 sure. Fixed that. Thank you.

**Comment by nlchar** [comment]
> @nlchar I think you forgot to revert back your changes to `app.eo`?

@yegor256 thanks, fixed that. Ready for review again.

**Comment by 0crat** [comment]
Job `gh:cqfn/eo#202` is not assigned, can't get performer

**Pull request closed without merging**